src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -482,44 +482,78 @@ public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+                //
+                // we can safely assume that this record is not the last one in the file, so let's just
+                // read the max Bytes required to store the largest structure , and compare results
+
+                int minDataDescriptorSize = 12;
+                if (reader.BaseStream.Length < reader.BaseStream.Position + extraFieldLength + entry.CompressedLength + minDataDescriptorSize)
                 {
                     return false;
                 }
 
                 reader.BaseStream.Seek(extraFieldLength + entry.CompressedLength, SeekOrigin.Current); // seek to end of compressed file from which Data descriptor starts
-                uint dataDescriptorSignature = reader.ReadUInt32();
-                bool wasDataDescriptorSignatureRead = false;
-                int seekSize = 0;
-                if (dataDescriptorSignature == DataDescriptorSignature)
+                Span<uint> buffer = stackalloc uint[6];
+
+                // let's try to match the smallest possible structure (3 x 4) 32 bit without signature
+                buffer[0] = reader.ReadUInt32();
+                buffer[1] = reader.ReadUInt32();
+                buffer[2] = reader.ReadUInt32();
+
+                if (TestMatch(entry, DataDescriptorSignature, buffer[0], buffer[1], buffer[2]))
                 {
-                    wasDataDescriptorSignatureRead = true;
-                    seekSize = 4;
+                    return true;
                 }
 
-                bool is64bit = entry._versionToExtract >= ZipVersionNeededValues.Zip64;
-                seekSize += (is64bit ? 8 : 4) * 2;   // if Zip64 read by 8 bytes else 4 bytes 2 times (compressed and uncompressed size)
+                // let's try to match the next record size (4 x 4) 32 bit with signature if another 4 byte is available
+                if (reader.BaseStream.Length < reader.BaseStream.Position + 4)
+                {
+                    return false;
+                }
+                buffer[3] = reader.ReadUInt32();
+                if (TestMatch(entry, buffer[0], buffer[1], buffer[2], buffer[3]))
+                {
+                    return true;
+                }
 
-                if (reader.BaseStream.Length < reader.BaseStream.Position + seekSize)
+                // At this point prior to trying to match 64 bit structures we need to make sure that version is high enough
+                if (entry._versionToExtract < ZipVersionNeededValues.Zip64)
                 {
                     return false;
                 }
 
-                // dataDescriptorSignature is optional, if it was the DataDescriptorSignature we need to skip CRC 4 bytes else we can assume CRC is alreadyskipped
-                if (wasDataDescriptorSignatureRead)
-                    reader.BaseStream.Seek(4, SeekOrigin.Current);
+                //let's try to match the 64 bit structures 64 bit without signature if another 4 byte is available
```

Comment:
```suggestion
                // let's try to match the 64 bit structures 64 bit without signature if another 4 byte is available
```

consistent formatting with similar comment above